cmd/controller/frontend/frontend.go

```diff
@@ -75,8 +151,15 @@ func (frontend *Frontend) getAgentById(id string) (restapi.Agent, error) {
 	return frontend.storage.GetAgentById(id)
 }
 
-func (frontend *Frontend) updateAgent(update restapi.AgentUpdate) error {
-	return frontend.storage.UpdateAgent(update)
+func (frontend *Frontend) updateAgent(ctx context.Context, update restapi.AgentUpdate) error {
+	err := frontend.storage.UpdateAgent(update)
+	if err == nil && len(update.Sessions) > 0 {
+		for sessionId, session := range update.Sessions {
+			frontend.webhookMessages <- fmt.Sprintf("/v1/session/update?agentId=%s&sessionId=%s&newState=%s", update.Id, sessionId, session.State)
```

Comment:
might need to URL Encode this?

Reply:
or send the data as POST data instead of a string?

Reply:
I was thinking of that as an alternative. The user provides a url, we post to that url with the session data. Seems cleaner?